src/plugins/intel_gpu/src/plugin/program_builder.cpp

```diff
@@ -103,6 +103,13 @@ ProgramBuilder::ProgramBuilder(std::shared_ptr<ov::Model> model, cldnn::engine&
     CustomLayer::LoadFromFile(custom_layers_config, m_custom_layers, custom_layers_config.empty());
 
     auto ops = model->get_ordered_ops();
+    // [TODO] We found a performance issue when multiple kernels are built as a single program.
+    // The below WA code needs to be removed after it is fixed.
```

Comment:
I'd like to leave a note that having reduced # of batch size make sense for dynamic models, because usually there are small # of kernels in dynamic models because most of the layers are mapped to a shape agnostic kernel. So having smaller # of batch size is even better for dynamic model as we can do more parallel build.

Reply:
I updated the comments as you reviewed. Thank you.